pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -639,25 +709,74 @@ func (r *ReconcileLogStorage) deleteInvalidECKManagedPublicCertSecret(ctx contex
-func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Context, instl *operatorv1.InstallationSpec) (oprKeyCert *corev1.Secret, esKeyCert *corev1.Secret, certSecret *corev1.Secret, err error) {
+// createKubeControllersSecrets checks for the existence of the secrets necessary for Kube controllers to access Elasticsearch through ES gateway and
+// creates them if they are missing. Kube controllers no longer uses admin credentials to make requests directly to Elasticsearch. Instead, gateway credentials
+// are generated and stored in the user secret, a hashed version of the credentials is stored in the tigera-elasticsearch namespace for ES Gateway to retrieve and use to compare
+// the gateway credentials, and a secret containing real admin level credentials is created and stored in the tigera-elasticsearch namespace to be swapped in once
+// ES Gateway has confirmed that the gateway credentials match.
+func (r *ReconcileLogStorage) createKubeControllersSecrets(ctx context.Context, kubeControllersGatewaySecret *corev1.Secret, esAdminUserSecret *corev1.Secret) (*corev1.Secret, *corev1.Secret, error) {
+	hashedPassword, err := bcrypt.GenerateFromPassword(kubeControllersGatewaySecret.Data["password"], bcrypt.MinCost)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	kubeControllersVerificationSecret, err := utils.GetSecret(ctx, r.client, render.ElasticsearchKubeControllersVerificationUserSecret, render.ElasticsearchNamespace)
+	if err != nil {
+		return nil, nil, err
+	}
+	if kubeControllersVerificationSecret == nil {
+		kubeControllersVerificationSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      render.ElasticsearchKubeControllersVerificationUserSecret,
+				Namespace: render.ElasticsearchNamespace,
+			},
+			Data: map[string][]byte{
+				"username": []byte(render.ElasticsearchKubeControllersUserName),
+				"password": hashedPassword,
+			},
+		}
+	}
+
+	kubeControllersSecureUserSecret, err := utils.GetSecret(ctx, r.client, render.ElasticsearchKubeControllersSecureUserSecret, render.ElasticsearchNamespace)
+	if err != nil {
+		return nil, nil, err
+	}
+	if kubeControllersSecureUserSecret == nil {
+		kubeControllersSecureUserSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      render.ElasticsearchKubeControllersSecureUserSecret,
+				Namespace: render.ElasticsearchNamespace,
+			},
+			Data: map[string][]byte{
+				"username": []byte("elastic"),
+				"password": esAdminUserSecret.Data["elastic"],
+			},
+		}
+	}
+
+	return kubeControllersVerificationSecret, kubeControllersSecureUserSecret, nil
+}
+
+// getESGatewayCertificateSecrets retrieves certificate secrets needed for ES Gateway to run or for
+// components to communicate with Elasticsearch/Kibana through ES Gateway. The order of the secrets returned are:
+// 1) The certificate secret needed for ES Gateway (in the operator namespace). If the user didn't create this it is created.
```

Comment:
the 2 comment is very clear. Could you include the same clarify for 1, by specifying where this is mounted and what it is used for? (I think, this secret is mounted into es-gateway for it to connect to ES/Kibana)

Reply:
Updated the comment.